### ISSUE TITLE
False positive R1730 consider-using-min-builtin, R1731 consider-using-max-builtin

### ISSUE THREAD

**Issue by bebound** [pylint-dev/pylint#9800]
### Bug description
The errors should not be raised.

```python
def kk(instance,count):
    if count>0:
        instance.count=count
    return instance

def kk2(instance,count):
    if count<0:
        instance.count=count
    return instance
```

### Command used

```shell
pylint --disable=C0114,C0116
```

### Pylint output

```python
test.py:2:4: R1730: Consider using 'count = min(count, count)' instead of unnecessary if block (consider-using-min-builtin)
test.py:7:4: R1731: Consider using 'count = max(count, count)' instead of unnecessary if block (consider-using-max-builtin)
```

### Expected behavior

No warning

### Pylint version

```shell
pylint 3.2.5
astroid 3.2.3
Python 3.10.12 (main, Mar 22 2024, 16:50:05) [GCC 11.4.0]
```

### OS / Environment

Ubuntu 22.04
